botocore/signers.py

```diff
@@ -575,7 +575,9 @@ def generate_presigned_url(self, ClientMethod, Params=None, ExpiresIn=3600,
 
     # Prepare the request dict by including the client's endpoint url.
     prepare_request_dict(
-        request_dict, endpoint_url=self.meta.endpoint_url)
+        request_dict, endpoint_url=self.meta.endpoint_url,
+        context={'protocol': self.meta.service_model.protocol}
```

Comment:
Yeah it is unfortunate that this appears to be really the only option we have given the signers do not receive any models in order to do the signing logic. One thing I do wonder is if we should put the `service_model` instead into the context? I only suggest this because the context could get a little bloated if we ever decide to add another handler that needs something other than the protocol from the model